doc/keepalive.md

```diff
@@ -0,0 +1,36 @@
+# Keepalive Ping User Guide for gRPC Core (and dependants)
+
+The keepalive ping is a way to check if a channel is currently working by sending HTTP2 pings over the transport. It is sent periodically, and if the ping is not acknowledged by the peer within a certain timeout period, the transport is disconnected.
+
+This guide documents the knobs within gRPC core to control the current behavior of the keepalive ping.
+
+The keepalive ping is controlled by two important channel arguments -
+* **GRPC_ARG_KEEPALIVE_TIME_MS**
+  * This channel argument controls the period (in milliseconds) after which a keepalive ping is sent on the transport.
+* **GRPC_ARG_KEEPALIVE_TIMEOUT_MS**
+  * This channel argument controls the amount of time (in milliseconds), the sender of the keepalive ping waits for an acknowledgement. If it does not receive an acknowledgement within this time, it will close the connection.
+
+The above two channel arguments should be sufficient for most users, but the following arguments can also be useful in certain use cases.
```

Comment:
Give an example like when someone needs a very short keepalive timeout. That seems to be the area where we have seen some confusion. I am ok without it though.

Reply:
In case of stream that keeps a database connection open, with the default keep alive that database connection will remain in a idle transaction status for around 2 hours.

Reply:
sure! depending on the usecase it might be a good idea to set these to something more desirable. Please refer https://github.com/grpc/proposal/blob/master/A8-client-side-keepalive.md for the original proposal